### DISCUSSION TITLE
Default vuepress theme's HomePage broken by adding seo2 plugin

### DISCUSSION THREAD

**Discussion by narumi147** [vuepress-theme-hope/vuepress-theme-hope#1023] [Q&A] [answered]
The default vuepress theme provide a home page with hero image, text and features intro when `home: true` is set in frontmatter. 

After I introduce the seo2 plugin, its dependency `vuepress-shared` plugin change the home page, the home page becomes a usual page with sidebar and original hero image, text, features intro are useless in this plugin's home setting.

I only want seo function here, is there any solution to disable the plugin's home theme?

**Comment by Mister-Hope** [comment]
I will have a look later. SEO do not have any client side, why would it break your layout?

Did it remove the original frontmatter? Any detailed information about how this plugin is effecting your client side is welcomed.

**Comment by narumi147** [comment]
![image](https://user-images.githubusercontent.com/56487872/134770389-ed84db8a-4643-4fc9-828f-7bbc0800e36e.png)
![image](https://user-images.githubusercontent.com/56487872/134770484-f0cd8eda-c2eb-470e-bf9d-63449509a0bf.png)

```md
---
home: true
title: Home
heroImage: /chaldea.png
heroText: Chaldea
actions:
  - text: Installation
    link: /installation.html
    type: primary
  - text: User Manual
    link: /introduction.html
    type: secondary
features:
  - title: Feature 1
    details: description 1.
  - title: Feature 1
    details: description 1.
  - title: Feature 1
    details: description 1.
  - title: Feature 1
    details: description 1.
    
footer: MIT Licensed | Copyright © 2021-present narumi
---
This is README.md
```

```ts
  plugins: [
    ['@vuepress/plugin-search'],
    ["vuepress-plugin-seo2"],
  ]
```


All frontmatter are removed, didn't render in html, I guess this plugin's default theme (in `shared`) has intercepted the `home` keyword.

**Reply by Mister-Hope** [reply]
Go it, I will checkout today.

**Reply by narumi147** [reply]
Another problem here. 

While only integrating sitemap plugin, building failed because of `themeConfig` is undefined when reading `themeConfig.git`. Although I don't know why need git plugin here? So inside plugins, it's better to check `themeConfig` exists or not before reading.

**Comment by narumi147** [comment]
for sitemap, i was using the legacy version `@mr-hope/vuepress-plugin-sitemap`, after migrating to `vuepress-plugin-sitemap2`, it works, though @next version cannot be used because it need vite.

**Reply by Mister-Hope** [reply]
That should be the actual problem, v1has different api with v2.

For plugin git, it's reading git database to get the creation time.

**Reply by Mister-Hope** [reply]
And for vite, my plugin do not care what bundler you are using, should be working on both bundler,  so could you explain why you think that "it need vite"

**Reply by narumi147** [reply]
for `vuepress-plugin-sitemap2`: `2.0.0-alpha.0` works, but `^2.0.0-alpha.5` failed
```
Error: Cannot find module 'vite'
Require stack:
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/vuepress-plugin-sitemap2/node_modules/@mr-hope/vuepress-shared/node/index.js
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/vuepress-plugin-sitemap2/node/index.js
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/@vuepress/core/lib/app/resolvePluginModule.js
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/@vuepress/core/lib/app/resolvePlugin.js
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/@vuepress/core/lib/app/appUse.js
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/@vuepress/core/lib/app/index.js
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/@vuepress/core/lib/index.js
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/@vuepress/cli/lib/commands/build/createBuild.js
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/@vuepress/cli/lib/commands/build/index.js
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/@vuepress/cli/lib/commands/index.js
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/@vuepress/cli/lib/index.js
- /Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/vuepress/bin/vuepress.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:94:18)
    at Object.<anonymous> (/Users/narumi/Projects/chaldea-project/chaldea-docs/node_modules/vuepress-plugin-sitemap2/node_modules/@mr-hope/vuepress-shared/node/index.js:1:75)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
error Command failed with exit code 1.
```

**Reply by narumi147** [reply]
Although we can build with `webpack` or `vite`, but if I only want to use webpack, and didn't add vite into `package.json`, the error will raise. 
So I need to add vite too to ensure the build works even I don't need it.

**Comment by narumi147** [comment]
Any progress on why vuepress' default `Home` is broken?

**Reply by Mister-Hope** [reply]
Yes, see a recent discussion in VuePress (by me)

There are some design issues.

**Reply by narumi147** [reply]
I see, is there any temp trick to bypass it?


Just link here to remind me.
- https://github.com/vuepress/vuepress-next/discussions/521

**Reply by Mister-Hope** [reply]
Yes, I am actually merging the default frontmatter in "a usual way", but it's not working as expected, some editions may needed at `@vuepress/core` to let this feature work well. I personally don't think I need to make some "hack" here, as other plugins may possibly break frontmatter:

https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/bfc530faf4d9f9950a183221c7e25e976467d414/packages/seo2/src/node/index.ts?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L36

Actually I can not reproduct the issue you are reporing in my workspace, so I guess your are hinting the situation I mentioned.

**Reply by narumi147** [reply]
Tried on master branch, still broken. Since I know nothing about node/js, maybe just wait for official fix. 😂
But it's strange why you cannot reproduce, what if test on the following repo? 
https://github.com/chaldea-center/chaldea-docs

**Comment by Mister-Hope** [comment, marked as answer]
beta 29 released and fix this issue by changing api(BREAKING CHANGES),  Once I build a compatblable version for it, it will be fine.
